python/src/opendp/extras/sklearn/linear_model/__init__.py

```diff
@@ -0,0 +1,118 @@
+'''
+This module requires extra installs: ``pip install 'opendp[scikit-learn]'``
+
+For convenience, all the functions of this module are also available from :py:mod:`opendp.prelude`.
+We suggest importing under the conventional name ``dp``:
+
+.. code:: python
+
+    >>> import opendp.prelude as dp
+
+The methods of this module will then be accessible at ``dp.sklearn.linear_model``.    
+
+If you're interested in the underlying algorithm, we've also
+`implemented Theil-Sen Regression as a demonstration of OpenDP plugins <../user-guide/plugins/theil-sen-regression.html>`_.
+'''
+
+from typing import Iterable
+from opendp.extras.sklearn.linear_model._make_private_theil_sen import (
+    make_private_theil_sen as _make_private_theil_sen,
+)  # noqa: F401
+from opendp._lib import import_optional_dependency
+
+
+class LinearRegression:
+    """
+    DP Linear Regression
+
+    The interface is parallel to that offered by sklearn's
+    `LinearRegression <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
+    The ``fit`` method returns an sklearn ``LinearRegression`` object.
+    """
+
+    def fit(
+        self,
+        X,
+        y,
+        x_bounds: Iterable[tuple[float, float]],
+        y_bounds: tuple[float, float],
+        scale: float,
+        runs: int = 1,
+        candidates_count: int = 100,
+        fraction_bounds: tuple[float, float] = (0.25, 0.75),
```

Comment:
I've thought more about the APIs and think it would be better to move these into the constructor.

This way arguments that would typically go to the constructor function are still a part of the constructor, but this time for the LinearRegression class. It would also be easier to integrate with the Context API, in that a user could ultimately do something like: `dp.query().sklearn(LinearRegression(x_bounds, y_bounds)).release()`.